src/utils/NumberUtils.h

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenColorIO Project.
+
+#ifndef INCLUDED_NUMBERUTILS_H
+#define INCLUDED_NUMBERUTILS_H
+
+#if defined(_MSC_VER)
+#define really_inline __forceinline
+#else
+#define really_inline inline __attribute__((always_inline))
+#endif
+
+#ifdef USE_FAST_FLOAT
+#include <fast_float/fast_float.h>
+#endif
+
+#include <cstdlib>
+#include <locale>
+#include <system_error>
+#include <type_traits>
+
+namespace OCIO_NAMESPACE
+{
+namespace NumberUtils
+{
+
+struct Locale
+{
+#ifdef _WIN32
+    Locale() : local(_create_locale(LC_ALL, "C"))
+    {
+    }
+    ~Locale()
+    {
+        _free_locale(local);
+    }
+    _locale_t local;
+#else
+    Locale() : local(newlocale(LC_ALL_MASK, "C", NULL))
+    {
+    }
+    ~Locale()
+    {
+        freelocale(local);
+    }
+    locale_t local;
+#endif
+};
+
+struct from_chars_result
+{
+    const char *ptr;
+    std::errc ec;
+};
+
+static const Locale loc;
```

Comment:
[Question] Do we care about having duplication of the global variable in each translation unit here?

Reply:
Hmm, I've seen this result in symbol clashes with very complex static library link trees. I can fix this by moving all implementation details to a cpp, and leaving a clean ABI stub here.

@hodoulp -- I think I now know why `StringUtils` is not defined inside `namespace OCIO_NAMESPACE`, that's already implied by every place where it's `#include`d. For the above to work, I'll have to define things in just `namespace NumberUtils`.

Reply:
Update. Doing so causes a conflict when building static OCIO, as CMake wants the helper to be a public library (and doesn't embed it inside OCIO itself).

Reply:
Hadn't thought about that, we can live it here if there is no issues anyone can see.